cumulusci/core/flowrunner.py

```diff
@@ -541,12 +541,8 @@ def _visit_step(
         if parent_ui_options is None:
             parent_ui_options = {}
 
-        # Step Validation
-        # - A step is either a task OR a flow.
-        if all(k in step_config for k in ("flow", "task")):
-            raise FlowConfigError(
-                f"Step {number} is configured as both a flow AND a task. \n\t{step_config}."
-            )
+        # This should never happen because of cleanup in core/utils/merge_config()
+        assert not all(step_type in step_config for step_type in ("task", "flow"))
```

Comment:
Can simplify this with a set:
```suggestion
        # This should never happen because of cleanup in core/utils/merge_config()
        assert step_config.keys() != {"task", "flow"}
```

Reply:
implemented